File: Programmers/1/12977. 소수 만들기/소수 만들기.cpp
#include <vector>
#include <iostream>
#include <math.h>

using namespace std;

int isPrime(int n)
{
    for (int i = 2; i <= sqrt(n); i++)
    {
        if (!(n % i))
            return 0;
    }
    
    return 1;
}

int solution(vector<int> nums) {
    int answer = 0;

    for (int i = 0; i < nums.size() - 2; i++)
    {
        for(int j = i + 1; j < nums.size() - 1; j++)
        {
            for(int k = j + 1; k < nums.size(); k++)
            {
                answer += isPrime(nums[i] + nums[j] + nums[k]);
            }
        }
    }

    return answer;
}

File: Programmers/1/12977. 소수 만들기/소수 만들기.java
import java.util.*;

class Solution {
    boolean[] prime = new boolean[3001];
    
    public void isPrime() {
        prime[0] = false;				
        prime[1] = false;

        for(int i = 2; i <= Math.sqrt(3000); i++) {
            if(prime[i] == true) {
                continue;
            }

            for(int j = i * i; j < prime.length; j = j + i) {
                prime[j] = true;
            }
        }
    }
    
    public int solution(int[] nums) {
        int answer = 0;
        
        isPrime();

        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                for (int k = j + 1; k < nums.length; k++) {
                    int num = nums[i] + nums[j] + nums[k];
                    
                    if (!prime[num]) {
                        answer++;
                    }
                }
            }
        }

        return answer;
    }
}